File: app/src/main/java/kashyap/genovatest/ui/gn/GNViewModel.java
package kashyap.genovatest.ui.gn;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

import java.io.IOException;
import java.io.InputStream;

public class GNViewModel extends ViewModel {

    private MutableLiveData<String> mText;

    public GNViewModel() {
        mText = new MutableLiveData<>();
        mText.setValue("This is home fragment");
    }

    public LiveData<String> getText() {
        return mText;
    }


}

File: app/src/main/java/kashyap/genovatest/ui/gn/GNViewModel.kt
package kashyap.genovatest.ui.gn

import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel

class GNViewModel : ViewModel() {
    private val mText: MutableLiveData<String>
    val text: LiveData<String>
        get() = mText

    init {
        mText = MutableLiveData()
        mText.value = "This is home fragment"
    }
}